File: Bab 8/Exercise/7. Pola String.c
#include <stdio.h>
#include <string.h>

int main()
{

	char string[99][99];
	int banyak, baris, kolom, panjang;

	panjang = 0;

	scanf("%d", &banyak);

	for (baris = 0; baris < banyak; baris++)
	{
		scanf("%s", &string[baris]);

		if (panjang < strlen(string[baris]))
		{
			panjang = strlen(string[baris]);
		}
	}


	for (baris = 0; baris < panjang; baris++)
	{
		for (kolom = 0; kolom < banyak; kolom++)
		{
		
			if (baris < strlen(string[kolom]))
			{
				if (kolom == banyak - 1)
				{
					printf("%c", string[kolom][baris]);
				}

				else{		

					printf("%c ", string[kolom][baris]);
				}	
			
			}else{

				printf("  ");
			}					
					
		}

		printf("\n");
	}
	
	return 0;
}

/*
11
hidup
adalah
anugerah
msti
kita
lakukan
yang
terbaik
dan
tetap
semangat
*/

File: Bab 8/Exercise/7. Pola String.java
import java.util.Scanner;

public class Main{

	public static void main(String[] args) {
		
		String[] string = new String[99];
		int banyak, baris, kolom, panjang;

		Scanner sc = new Scanner(System.in);

		banyak = 0;
		baris = 0;
		kolom = 0;
		panjang = 0;

		try{

			banyak = sc.nextInt();

		}catch(Exception e){

		}

		for (baris = 0; baris < banyak; baris++) {

			try{

				string[baris] = sc.next();

			}catch(Exception e){

			}

			if (panjang < string[baris].length())
			{
				panjang = string[baris].length();
			}
		}

		for (baris = 0; baris < panjang; baris++)
		{
			for (kolom = 0; kolom < banyak; kolom++)
			{
				if (baris < string[kolom].length()) {
				
					if (kolom == banyak - 1)
					{
						System.out.print(string[kolom].charAt(baris));
					}

					else{		

						System.out.print(string[kolom].charAt(baris) + " ");
					}		

				}else{

					System.out.print("  ");
				}			
			}

			System.out.println();
		}
	}
}